From b8e697ab1017a2798ea9dd5a05fe864827a18165 Mon Sep 17 00:00:00 2001
From: Ethan Chen <intervigil@gmail.com>
Date: Thu, 15 Oct 2015 13:50:10 -0700
Subject: [PATCH 18/27] Revert "egl: Remove old tuna BGRA vs RGBA workaround."

chrono: move WORKAROUND_BUG_10194508 define to cpp file.
Thanks SergeyL@4pda.ru for pointing out to this patch.

This reverts commit 733a80754786d39cdc0fee09509b194472c320bc.

Change-Id: Ifb41720ed020489892a667914ea3bd3f1ac6504e
---
 opengl/libs/EGL/eglApi.cpp | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/opengl/libs/EGL/eglApi.cpp b/opengl/libs/EGL/eglApi.cpp
index ba3a5f9..b46600f 100644
--- a/opengl/libs/EGL/eglApi.cpp
+++ b/opengl/libs/EGL/eglApi.cpp
@@ -15,6 +15,7 @@
  */
 
 #define ATRACE_TAG ATRACE_TAG_GRAPHICS
+#define WORKAROUND_BUG_10194508
 
 #include <ctype.h>
 #include <dlfcn.h>
@@ -503,6 +504,15 @@ EGLSurface eglCreateWindowSurface(  EGLDisplay dpy, EGLConfig config,
 
         EGLint format;
         android_dataspace dataSpace = HAL_DATASPACE_UNKNOWN;
+
+#ifdef WORKAROUND_BUG_10194508
+        if (!cnx->egl.eglGetConfigAttrib(iDpy, config, EGL_NATIVE_VISUAL_ID,
+                &format)) {
+            ALOGE("eglGetConfigAttrib(EGL_NATIVE_VISUAL_ID) failed: %#x",
+                    eglGetError());
+            format = 0;
+        }
+#else
         EGLint a = 0;
         EGLint r, g, b;
         r = g = b = 0;
@@ -537,6 +547,7 @@ EGLSurface eglCreateWindowSurface(  EGLDisplay dpy, EGLConfig config,
                 format = HAL_PIXEL_FORMAT_RGBA_FP16;
             }
         }
+#endif
 
         // now select a corresponding sRGB format if needed
         if (attrib_list && dp->haveExtension("EGL_KHR_gl_colorspace")) {
-- 
2.7.4

